### PR TITLE
Add a new permissionsRequired option to a manifest's object. Fixes STCON-68

### PR DESCRIPTION
**Pull request by mkuklis**
https://issues.folio.org/browse/STCON-68

This PR introduces a new `permissionsRequired` prop to a manifest's object. The initial discussion can be found here:

https://github.com/folio-org/ui-users/commit/8558adef98e1d135314eaf9831d2decc459021b0#diff-6c4a67641cbff8412246381fb1851b26

### CODE REVIEW

File: RESTResource/RESTResource.js [hunk 1]
```diff
@@ -337,6 +337,12 @@ export default class RESTResource {
     this.dispatch(this.fetchAction(this.cachedProps));
   }
 
+  hasPerm(state, perms) {
+    const currentPerms = _.get(state, ['okapi', 'currentPerms'], {});
+    const reqPerms = _.isArray(perms) ? perms : perms.split(',');
+    return !reqPerms.find(perm => !currentPerms[perm]);
```
Comment (MikeTaylor) [inline RESTResource/RESTResource.js]: This is an elegant line, though I did have to look at it a couple of times before I got it.

File: RESTResource/RESTResource.js [hunk 2]
```diff
@@ -471,8 +477,16 @@ export default class RESTResource {
...
   fetchAction = (props) => {
     const key = this.stateKey();
+
     return (dispatch, getState) => {
-      const options = this.verbOptions('GET', getState(), props);
+      const state = getState();
+      const options = this.verbOptions('GET', state, props);
+
+      if (options.permissionsRequired && !this.hasPerm(state, options.permissionsRequired)) {
+        dispatch(this.actions.fetchAbort({ message: 'cannot satisfy request: missing permissions' }));
```
Comment (MikeTaylor) [inline RESTResource/RESTResource.js]: Maybe include the name of the missing permissions in the message?

**Comment by mkuklis** [comment]
@MikeTaylor thank you for taking a look at it!

File: RESTResource/RESTResource.js (hunk 1 above)
Comment (mkuklis) [reply RESTResource/RESTResource.js]: @MikeTaylor thanks. I tried to find a one liner for this. Not the easiest to read though :/

File: RESTResource/RESTResource.js (hunk 2 above)
Comment (mkuklis) [reply RESTResource/RESTResource.js]: Nice idea @MikeTaylor. I think it will complicate the `hasPerm` a bit but maybe worth it.

**Comment by doytch** [comment]
Nice nice. Love it!

**Pull request merged**